File: pinyougou_parent/pinyougou_seckill_web/src/main/webapp/js/service/cartService.js
app.service("cartService",function ($http) {
    //查询购物车列表
    this.findCartList=function () {
        return $http.get("cart/findCartList.do");
    }
    //商品数量加减操作
    this.addItemToCartList=function (itemId,num) {
        return $http.get("cart/addItemToCartList.do?itemId="+itemId+"&num="+num);
    }
});

File: pinyougou_parent/pinyougou_cart_interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import groupEntity.Cart;

import java.util.List;

public interface CartService {
    /**
     * 添加商品到购物车列表
     */
    public List<Cart> addItemToCartList(List<Cart> cartList,Long itemId,Integer num);

    void saveCartListToRedisBySessionId(String sessionId, List<Cart> cartList);

    List<Cart> findCartListFromRedis(String sessionId);

    void saveCartListToRedisByUserName(String username, List<Cart> cartList);

    List<Cart> mergeCartList(List<Cart> cartList_sessionId, List<Cart> cartList_username);

    void deleteCartList(String sessionId);
}
